subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/execution/PartialEvaluator.kt

```diff
@@ -20,12 +20,24 @@ import org.jetbrains.kotlin.lexer.KotlinLexer
 import org.jetbrains.kotlin.lexer.KtTokens
 
 
+enum class ProgramKind {
+    TopLevel,
+    ScriptPlugin
+}
+
+
+enum class ProgramTarget {
+    Project,
+    Settings
+}
```

Comment:
Doesn't this limit scripts that are applied to a different target type?
```
apply "foo.gradle.kts" to something
```

Reply:
For now only `Project` and `Settings` are supported.

In general, if you apply the same script to different targets, the interpreter will specialise the program once for each target.

Reply:
I understand that only `Project` and `Settings` is currently supported, I just want to make sure that this code isn't somehow limiting future support for other targets.

Also, how does this handle one script that is both targeted at being a `Settings` and a `Project` target. This is a setup I have in my repo today and I want to make sure that this use case isn't broken by this.

Reply:
> I just want to make sure that this code isn't somehow limiting future support for other targets.

Rest assured it isn't :)

> This is a setup I have in my repo today and I want to make sure that this use case isn't broken by this.

It should continue to work as before.